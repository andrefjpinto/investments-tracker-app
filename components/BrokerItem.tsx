import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import Broker from '../models/broker'

import Card from './UI/Card'

type BrokerItemProps = {
  broker: Broker
}

const BrokerItem = (props : BrokerItemProps) => {
  return (
    <Card style={styles.brokerItem}>
      <Text style={styles.title}>{props.broker.Name}</Text>
      <Text style={styles.subTitle}>{props.broker.URL}</Text>
    </Card>
  )
}

const styles = StyleSheet.create({
  brokerItem: {
    marginHorizontal: 20,
    marginVertical: 10,
    padding: 10,
    alignItems: 'center'
  },
  title: {
    fontSize: 18
  },
  subTitle: {
    fontSize: 12
  }
})

export default BrokerItem
