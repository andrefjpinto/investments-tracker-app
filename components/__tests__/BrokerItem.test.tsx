import React from 'react'
import renderer from 'react-test-renderer'

import BrokerItem from '../BrokerItem'

describe('BrokerItem', () => {
  const broker = { Name: 'Test Broker', URL: 'https://test.broker' }

  it('renders correctly', () => {
    const tree = renderer.create(<BrokerItem broker={broker} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})