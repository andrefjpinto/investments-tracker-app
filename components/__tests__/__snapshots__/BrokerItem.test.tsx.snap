// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrokerItem renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 10,
      "elevation": 5,
      "marginHorizontal": 20,
      "marginVertical": 10,
      "padding": 10,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.26,
      "shadowRadius": 8,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
      }
    }
  >
    Test Broker
  </Text>
  <Text
    style={
      Object {
        "fontSize": 12,
      }
    }
  >
    https://test.broker
  </Text>
</View>
`;
